### PR TITLE
feat(): initiative template entity

### PR DESCRIPTION
**Pull request by jordantsanz**
1. Description:
Creates the Hub Initiative Template entity.
1. Instructions for testing:

1. Closes Issues:[ #7276](https://zentopia.esri.com/workspaces/collaboration-sprint-board-614f9bfc0946c40011ef574e/issues/dc/hub/7276)

1. [ ] Updated meaningful TSDoc to methods including Parameters and Returns, see [Documentation Guide](https://esri.github.io/hub-components/storybook/?path=/story/guides-documentation--page)

1. [X] used semantic commit messages
  
1. [X] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

1. [ ] updated `peerDependencies` as needed. **CRITICAL** our automated release system can **not** be counted on to update `peerDependencies` so we _must_ do it manually in our PRs when needed. See the [updating peerDependencies](/RELEASE.md#Updating-peerDependencies) section of the release instructions for more details.

### CODE REVIEW

**Review by juliannemarik — COMMENTED** [review]
Did a quick sweep through and left some minor comments.

File: packages/common/src/initiativeTemplates/_internal/computeProps.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * Given a model and an initiative template, set various computed properties that can't be directly mapped
+ * @private
+ * @param model
+ * @param initiativeTemplate
+ * @param requestOptions
+ * @returns
+ */
+export function computeProps(
```
[the remark this answers is not in the document]
Comment (jordantsanz) [reply packages/common/src/initiativeTemplates/_internal/computeProps.ts]: That makes sense to me -- once your PR lands I will refactor to use the computeLinks logic 👍

File: packages/common/src/core/types/IHubInitiativeTemplate.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+/** Defines the properties of a Hub Initiative Template object */
+export interface IHubInitiativeTemplate
+  extends IHubItemEntity,
+    IWithSlug,
+    IWithCatalog,
+    IWithLayout {
+  previewUrl: string;
+  recommendedTemplates: string[];
+  siteSolutionId: string;
```
Comment (dbouwman) [inline packages/common/src/core/types/IHubInitiativeTemplate.ts:13]: Now is the time to consider renaming these properties... not saying we need to

**Review by dbouwman — APPROVED** [review]
Did not to a deep dive through all the details, but this looks like a good first cut.

**Pull request merged**